torch/_inductor/decomposition.py

```diff
@@ -371,6 +371,13 @@ def baddbmm(self, batch1, batch2, beta=1, alpha=1):
     return self + result
 
 
+@register_decomposition([aten.cat.default])
+def cat(tensors, dim=0):
+    if len(tensors) == 1:
```

Comment:
we already have this special case in `cat` lowering, why do we need it in decomposition?

Reply:
The issue fixed in this PR is `pattern_matcher.py` is matching a mm+cat fusion and that fused pattern didn't handle this case.  So it wasn't doing through the normal lowering.

Reply:
But I've just tried and original issue is fixed just by changes in `patter_matcher.py`, this change is not required, `cat_tuned_op` is used in lowering patterns only, they never go through decompositions I thought?

Reply:
Correct.  Though matching that pattern can prevent other patterns from getting matched (for example an mm to addmm), so it is good to remove the pointless cat from the graph.